src/pages/workspaces/workspace/analysis/utils/tool-utils.ts

```diff
@@ -176,8 +176,8 @@ export const allAppTypes: AppToolLabel[] = _.flow(_.map('label'), _.compact)(app
 
 export const isPauseSupported = (toolLabel: ToolLabel): boolean => !_.find((tool: AppTool | RuntimeTool) => tool.label === toolLabel)(tools)?.isPauseUnsupported
 
-export const isSettingsSupported = (toolLabel: ToolLabel, cloudProvider: CloudProvider, accessLevel: string, createdDate: string): boolean => Utils.cond(
-  [toolLabel === appToolLabels.CROMWELL && cloudProvider === cloudProviders.azure.label, () => doesWorkspaceSupportCromwellApp(createdDate, cloudProvider, toolLabel) && isOwner(accessLevel)],
+export const isSettingsSupported = (toolLabel: ToolLabel, cloudProvider: CloudProvider, createdBy: string, createdDate: string): boolean => Utils.cond(
+  [toolLabel === appToolLabels.CROMWELL && cloudProvider === cloudProviders.azure.label, () => doesWorkspaceSupportCromwellApp(createdDate, cloudProvider, toolLabel) && createdBy === getUser()?.email],
```

Comment:
looks like `doesWorkspaceSupportCromwell` should take the `WorkspaceInfo` as well and include this boolean comparison since it already returns a boolean. Don't like addiing another condition here

Reply:
I just realized that combining the logic in `doesWorkspaceSupportCromwellApp` **might** solve the bug mentioned in https://broadworkbench.atlassian.net/browse/WM-1942. I will try this out